dev/a11y_assessments/lib/use_cases/material_banner.dart

```diff
@@ -24,48 +24,62 @@ class MainWidget extends StatefulWidget {
   @override
   State<MainWidget> createState() => MainWidgetState();
 }
-
 class MainWidgetState extends State<MainWidget> {
-  double currentSliderValue = 20;
-  ScaffoldFeatureController<MaterialBanner, MaterialBannerClosedReason>? controller;
+
+  FocusNode dismissButtonFocusNode = FocusNode();
+  FocusNode showButtonFocusNode = FocusNode();
+
+  @override
+  void dispose() {
+    dismissButtonFocusNode.dispose();
+    showButtonFocusNode.dispose();
+    super.dispose();
+  }
+
+  void hideBanner() {
+    ScaffoldMessenger.of(context).hideCurrentMaterialBanner();
+    showButtonFocusNode.requestFocus();
+  }
+
+  void showBanner() {
+    ScaffoldMessenger.of(context).showMaterialBanner(
+      MaterialBanner(
+        padding: const EdgeInsets.all(20),
+        content: const Text('Hello, I am a Material Banner'),
+        leading: const Icon(Icons.agriculture_outlined),
+        backgroundColor: Colors.blue,
+        actions: <Widget>[
+          TextButton(
+            focusNode: dismissButtonFocusNode,
+            onPressed: () {
+              setState(() {
+                hideBanner();
+              });
+            },
+            child: const Text('DISMISS'),
+          ),
+        ],
+      ),
+    );
+    dismissButtonFocusNode.requestFocus();
+  }
 
   @override
   Widget build(BuildContext context) {
-    VoidCallback? onPress;
-    if (controller == null) {
-      onPress = () {
-        setState(() {
-          controller = ScaffoldMessenger.of(context).showMaterialBanner(
-            MaterialBanner(
-              padding: const EdgeInsets.all(20),
-              content: const Text('Hello, I am a Material Banner'),
-              leading: const Icon(Icons.agriculture_outlined),
-              backgroundColor: Colors.green,
-              actions: <Widget>[
-                TextButton(
-                  onPressed: () {
-                    controller!.close();
-                    setState(() {
-                      controller = null;
-                    });
-                  },
-                  child: const Text('DISMISS'),
-                ),
-              ],
-            ),
-          );
-        });
-      };
-    }
     return Scaffold(
       appBar: AppBar(
         backgroundColor: Theme.of(context).colorScheme.inversePrimary,
         title: const Text('MaterialBanner'),
       ),
       body: Center(
         child: ElevatedButton(
-          onPressed: onPress,
+          focusNode: showButtonFocusNode,
```

Comment:
Same here, the setState is not needed.
Also, the `child` should go last per https://dart.dev/tools/linter-rules/sort_child_properties_last

```suggestion
ElevatedButton(
  focusNode: showButtonFocusNode,
  onPressed: showBanner,
  child: const Text('Show a MaterialBanner'),
)
```